multimodal/src/autogluon/multimodal/utils/inference.py

```diff
@@ -113,3 +111,37 @@ def infer_batch(
                 output = model_postprocess_fn(output)
 
     return output[model.prefix]
+
+
+def use_realtime(data: pd.DataFrame, data_processors: Dict):
+    """
+    Determine whether to use the realtime inference based on the sample number
+    and the data modalities. Loading image data requires more time than text.
+    Thus, we set a small threshold for image data. We may also consider the
+    model size in future, but we need to ensure this function is efficient since
+    using this function also costs additional inference time.
+
+    Parameters
+    ----------
+    data
+        A dataframe.
+    data_processors
+        A dict of data processors.
+
+    Returns
+    -------
+    Whether to use the realtime inference.
+    """
+    realtime = False
+    sample_num = len(data)
+    if IMAGE in data_processors and len(data_processors[IMAGE]) > 0:  # has image
+        if sample_num <= 10:
```

Comment:
How about to also detect the eval_batch_size in config? https://github.com/awslabs/autogluon/blob/c6d14b81402c35d4efe895201f0407f816714a00/multimodal/src/autogluon/multimodal/configs/environment/default.yaml#L6

We can take the minimum of two values: eval_batch_size, and recommended max sample_num for triggering real-time inference.

Reply:
I think `10` used here is independent of the `eval_batch_size `. I chose 10 here based on the time comparisons between the realtime inference and the default inference. 10 is a relatively safe number to guarantee that the realtime inference is faster than the default inference for image models.

Reply:
I'm suggesting logic like `min(eval_batch_size, 10)`. This will help us support larger-scale models (e.g., T0-11B).

Reply:
Make sense. Will improve the robustness in the next PR.